modules/installation-azure-service-principal.adoc

```diff
@@ -0,0 +1,124 @@
+    "name": "you@example.com",
+    "type": "user"
+  }
+}
+----
+<1> Ensure that the value of the name parameter is the name of your subscription.
+
+.. You can change the subscription to use by obtaining the subscription ID
+from the Azure portal and setting it in the CLI:
++
+----
+$ az account set -s <subscription_ID>
+----
+
+.. If you changed the subscription ID, display your account information again:
++
+----
+$ az account show
+
+{
+  "environmentName": "AzureCloud",
+  "id": "33212d16-bdf6-45cb-b038-f6565b61edda",
+  "isDefault": true,
+  "name": "Subscription Name",
+  "state": "Enabled",
+  "tenantId": "8049c7e9-c3de-762d-a54e-dc3f6be6a7ee",
+  "user": {
+    "name": "you@example.com",
+    "type": "user"
+  }
+}
+----
+
+. Record the values of the `tenantId` and `id` parameters from the previous
+output. You need these values during {product-title} installation.
+
+. Create the service principal for your account:
++
+----
+$ az ad sp create-for-rbac --role "User Access Administrator" -role Contributor --name <service_principal> <1>
+Changing "<service_principal>" to a valid URI of "http://<service_principal>", which is the required format used for service principal names
+Retrying role assignment creation: 1/36
+Retrying role assignment creation: 2/36
+Retrying role assignment creation: 3/36
+Retrying role assignment creation: 4/36
+{
+  "appId": "8bd0d04d-0ac2-43a8-928d-705c598c6956",
+  "displayName": "<service_principal>",
+  "name": "http://<service_principal>",
+  "password": "ac461d78-bf4b-4387-ad16-7e32e328aec6",
+  "tenant": "6048c7e9-b2ad-488d-a54e-dc3f6be6a7ee"
+}
+
+----
+<1> Specify the name to assign to the service principal.
+
+. Record the values of the `appId` and `password` parameters from the previous
+output. You need these values during {product-title} installation.
+
+. Grant additional permissions to the service principal. The service principal
+requires the legacy `Azure Active Directory Graph -> Application.ReadWrite.OwnedBy`
+permission and the `User Access Administrator` role for the cluster to
+credentials for its components.
+.. To assign the `User Access Administrator` role, run the following command:
++
+----
+$  az role assignment create --role "User Access Administrator" \
+    --assignee-object-id $(az ad sp list --filter "appId eq '<appId>'" \ <1>
+       | jq '.[0].objectId' -r)
+----
+<1> Specify the `appId` parameter value for your service principal.
+
+.. To assign the `Azure Active Directory Graph` permission, run the following
+command:
++
+----
+$ az ad app permission add --id <appId> \ <1>
+     --api 00000002-0000-0000-c000-000000000000 \
+     --api-permissions 824c81eb-e3f8-4ee6-8f6d-de7f50d565b7=Role
+----
+<1> Specify the `appId` parameter value for your service principal.
+.. If your account does not have the Azure Active Directory tenant administrator
+role, follow the guidelines for your organization to request that the tenant
+administrator approve your permissions request.
```

Comment:
`az ad app permission add` only creates a request. It still has to be granted by the tenant administrator. Not sure if it's granted right away for account with `Azure Active Directory tenant administrator`.

Reply:
Pretty sure it's *not* automatically granted.